android/src/main/java/io/invertase/firebase/notifications/RNFirebaseBackgroundNotificationsService.java

```diff
@@ -0,0 +1,32 @@
+package io.invertase.firebase.notifications;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+
+import com.facebook.react.HeadlessJsTaskService;
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.jstasks.HeadlessJsTaskConfig;
+import com.google.firebase.messaging.RemoteMessage;
+
+import javax.annotation.Nullable;
+
+import io.invertase.firebase.messaging.MessagingSerializer;
+
+public class RNFirebaseBackgroundNotificationsService extends HeadlessJsTaskService {
+  @Override
+  protected @Nullable HeadlessJsTaskConfig getTaskConfig(Intent intent) {
+    Bundle extras = intent.getExtras();
+    if (extras != null) {
+      WritableMap notificationMap = Arguments.makeNativeMap(intent.getExtras());
```

Comment:
We need to make sure that the object being passed to the JS side conforms to the `NotificationOpen` structure.  At the moment it doesn't allow for a `snooze` action with a remote input specified.

Take a look at the `parseIntentForLocalNotification` method in `RNFirebaseNotifications` to see how this is done.

Reply:
Done

Reply:
👍